File: src/main/java/com/dsm/pick/domains/domain/Student.java
package com.dsm.pick.domains.domain;

import javax.persistence.*;

@Entity
@Table(name = "student")
public class Student {

    @Id
    @Column(name = "num")
    private String num;

    @Column(name = "name")
    private String name;

    @ManyToOne
    @JoinColumn(name = "club_name", referencedColumnName = "name")
    private Club club;

    @ManyToOne
    @JoinColumn(name = "class_name", referencedColumnName = "name")
    private SchoolClass schoolClass;

    public Student() {}
    public Student(String num, String name, Club club, SchoolClass schoolClass) {
        this.num = num;
        this.name = name;
        this.club = club;
        this.schoolClass = schoolClass;
    }

    public String getNum() {
        return num;
    }

    public void setNum(String num) {
        this.num = num;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Club getClub() {
        return club;
    }

    public void setClub(Club club) {
        this.club = club;
    }

    public SchoolClass getSchoolClass() {
        return schoolClass;
    }

    public void setSchoolClass(SchoolClass schoolClass) {
        this.schoolClass = schoolClass;
    }
}

File: src/main/kotlin/com/dsm/pick/domain/Student.kt
package com.dsm.pick.domain

import javax.persistence.*

@Entity
@Table(name = "student")
class Student(

    @Id
    @Column(name = "num")
    val number: String,

    @Column(name = "name")
    val name: String,

    @ManyToOne
    @JoinColumn(name = "club_name", referencedColumnName = "name")
    val club: Club,

    @ManyToOne
    @JoinColumn(name = "class_name", referencedColumnName = "name")
    val classroom: Classroom,

    @Column(name = "is_monday_self_study")
    val isMondaySelfStudy: Boolean,

    @Column(name = "is_tuesday_self_study")
    val isTuesdaySelfStudy: Boolean,
) {

//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//
//        other as Student
//
//        if (number != other.number) return false
//        if (name != other.name) return false
//
//        return true
//    }
//
//    override fun hashCode(): Int {
//        var result = number.hashCode()
//        result = 31 * result + name.hashCode()
//        return result
//    }
}